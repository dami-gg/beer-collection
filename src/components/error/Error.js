// @flow
import React, { PureComponent } from "react";

import error from "../../assets/images/error.png";

import "./error.scss";

type Props = {};

class Error extends PureComponent<Props> {
  render() {
    return (
      <div className="error-page">
        <h1 className="error-page__title">This page could not be found!</h1>
        <div className="error-page__image-wrapper">
          <img className="error-page__image" src={error} alt="error" />
        </div>
        <p className="error-page__text">Please try again</p>
      </div>
    );
  }
}

export default Error;
