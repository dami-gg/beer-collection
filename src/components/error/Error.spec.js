import React from "react";
import Error from "./Error";

import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("Error", () => {
  let component;

  it("should render without crashing", () => {
    shallow(<Error />);
  });

  it("should match the snapshot", () => {
    component = renderer.create(<Error />);
    let json = component.toJSON();

    expect(json).toMatchSnapshot();
  });
});
