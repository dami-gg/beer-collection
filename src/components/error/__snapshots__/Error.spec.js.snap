// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should match the snapshot 1`] = `
<div
  className="error-page"
>
  <h1
    className="error-page__title"
  >
    This page could not be found!
  </h1>
  <div
    className="error-page__image-wrapper"
  >
    <img
      alt="error"
      className="error-page__image"
      src="error.png"
    />
  </div>
  <p
    className="error-page__text"
  >
    Please try again
  </p>
</div>
`;
